packages/xml-views-completion/src/api.ts

```diff
@@ -24,9 +29,40 @@ export function getXMLViewCompletions(
   });
 
   const allowedVisibility: UI5Visibility[] = ["public", "protected"];
-  const publicAndProtectedSuggestions = filter(suggestions, _ =>
-    includes(allowedVisibility, _.ui5Node.visibility)
+  const publicAndProtectedSuggestions = filter(
+    suggestions,
+    _ =>
+      isLiteralXMLViewCompletion(_) ||
+      includes(allowedVisibility, _.ui5Node.visibility)
   );
 
   return publicAndProtectedSuggestions;
 }
+
+export function isLiteralXMLViewCompletion(
```

Comment:
could we model the boolean as a UI5 node and avoid having a different structure causing multiple new branches of the already complex code?

could not a boolean "true" be represented a "UI5PrimitiveBooleanLiteral" with `name` "true" ?
and no description / deprecated info / additional info.

Reply:
The UI5 node could be dynamically created.

Reply:
Can you think of another way to model this so there are not two different structures?
Its fine to even refactor the existing flows.

Random thoughts:
- Could/Should suggestions include methods and not only data (I am not fond of this but maybe it would help?)
- Would we have need more than a simple string(attribute) or identifier(tag) literal suggestion? even in a binding expression you normally only offer the next identifier not a FQN (at least in most languages).

Reply:
It doesn't sound right to me to force literals to be UI5 nodes... they aren't. They're not part of the semantic model, don't exist on a namespace or have a parent, don't have a corresponding definition or exist in the api.json in any way.
We can model the literal "node" to look a bit like a UI5 node (put the `name` , `type` and `value` in a structure instead of being separate) which could make the code a bit simpler (though not by much - all the documentation for example would still have to be handled separately). We'll have to change the property name on the suggestion from `ui5Node` because it would no longer be a ui5 node.
If we want to change the structure to be completely agnostic to UI5 nodes we could return `CompletionItem`s... (I don't think this is a good idea, but it would take care of the difference in structures).